src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -1724,4 +1733,250 @@ void scaleInletFlows(EnergyPlusData &state, int const zoneNodeNum, int const spa
+                              this->Name));
+        break;
+    default:
+        break;
+    }
+
+    if (spacesTotal < 0.00001) {
+        ShowSevereError(state,
+                        format("ZoneEquipmentSplitterMixer::size: Total {} is zero for {}={}. Unable to autosize Space Fractions.",
+                               DataZoneEquipment::spaceEquipSizingBasisNamesUC[(int)this->spaceSizingBasis],
+                               BranchNodeConnections::ConnectionObjectTypeNames[(int)this->spaceEquipType],
+                               this->Name));
+        Real64 spaceFrac = 1.0 / (int)this->spaces.size();
+        ShowContinueError(state, format("Setting space fractions to 1/number of spaces = {}.", spaceFrac));
+        for (auto &thisSpace : this->spaces) {
+            thisSpace.fraction = spaceFrac;
+        }
+        return;
+    }
+
+    // Calculate space fractions
+    int spaceCounter = 0;
+    for (auto &thisSpace : this->spaces) {
+        ++spaceCounter;
+        if (thisSpace.fraction == DataSizing::AutoSize) {
+            switch (this->spaceSizingBasis) {
+            case DataZoneEquipment::SpaceEquipSizingBasis::DesignCoolingLoad:
+                thisSpace.fraction = state.dataSize->FinalSpaceSizing(thisSpace.spaceIndex).DesCoolLoad / spacesTotal;
+                break;
+            case DataZoneEquipment::SpaceEquipSizingBasis::DesignHeatingLoad:
+                thisSpace.fraction = state.dataSize->FinalSpaceSizing(thisSpace.spaceIndex).DesHeatLoad / spacesTotal;
+                break;
+            case DataZoneEquipment::SpaceEquipSizingBasis::FloorArea:
+                thisSpace.fraction = state.dataHeatBal->space(thisSpace.spaceIndex).FloorArea / spacesTotal;
+                break;
+            case DataZoneEquipment::SpaceEquipSizingBasis::Volume:
+                thisSpace.fraction = state.dataHeatBal->space(thisSpace.spaceIndex).Volume / spacesTotal;
+                break;
+            case DataZoneEquipment::SpaceEquipSizingBasis::PerimeterLength:
+                ShowFatalError(state,
+                               format("ZoneEquipmentSplitterMixer::size: Space Fraction Method={} not supported for {}={}",
+                                      DataZoneEquipment::spaceEquipSizingBasisNamesUC[(int)this->spaceSizingBasis],
+                                      BranchNodeConnections::ConnectionObjectTypeNames[(int)this->spaceEquipType],
+                                      this->Name));
+                break;
+            default:
+                break;
+            }
+            BaseSizer::reportSizerOutput(state,
+                                         BranchNodeConnections::ConnectionObjectTypeNames[(int)this->spaceEquipType],
+                                         this->Name,
+                                         format("Space {} Fraction", spaceCounter),
+                                         thisSpace.fraction);
+        }
+    }
+}
```

Comment:
Could combine these 2 switch blocks. For another day.

Reply:
Hmm, the motivation to separate them was to calculate `spacesTotal` first, then check to see if it's zero before using it to divide. But I'm open to other suggestions (on another day).